ospd_openvas/messaging/mqtt.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) 2021 Greenbone Networks GmbH
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import logging
+
+from functools import partial
+from typing import Callable, Type
+
+import paho.mqtt.client as mqtt
+
+from ..messages.message import Message
+
+from .publisher import Publisher
+from .subscriber import Subscriber
+
+logger = logging.getLogger(__name__)
+
+OSPD_OPENVAS_MQTT_CLIENT_ID = "ospd-openvas"
+
+QOS_AT_LEAST_ONCE = 1
+
+
+class MQTTClient(mqtt.Client):
+    def __init__(
+        self,
+        mqtt_broker_address: str,
+        mqtt_broker_port: int,
+        client_id=OSPD_OPENVAS_MQTT_CLIENT_ID,
+    ):
+        self._mqtt_broker_address = mqtt_broker_address
+        self._mqtt_broker_port = mqtt_broker_port
+
+        super().__init__(client_id=client_id, protocol=mqtt.MQTTv5)
+
+        self.enable_logger()
+
+    def connect(
+        self,
+        host=None,
+        port=None,
+        keepalive=60,
+        bind_address="",
+        bind_port=0,
+        clean_start=mqtt.MQTT_CLEAN_START_FIRST_ONLY,
+        properties=None,
+    ):
+        if not host:
+            host = self._mqtt_broker_address
+        if not port:
+            port = self._mqtt_broker_port
+
+        return super().connect(
+            host,
+            port=port,
```

Comment:
```suggestion
            port=int(port), // because of config, feeling cute, might delete later.
```

Reply:
The parser already parses network ports to int:
```py
    def network_port(self, string: str) -> int:
        """Check if provided string is a valid network port."""

        value = int(string)
        if not 0 < value <= 65535:
            raise argparse.ArgumentTypeError(
                'port must be in ]0,65535] interval'
            )
        return value
```

Reply:
yeah but also if gathered from config file?

Reply:
At least for me :trollface: 
I updated it, so you can test it